packages/squiggle-lang/src/dist/SymbolicDist.ts

```diff
@@ -854,6 +855,103 @@ export class Bernoulli extends SymbolicDist {
     );
   }
 }
+export class Metalog extends SymbolicDist {
+  a: number[];
+  private constructor({ a }: { a: number[] }) {
+    super();
+    this.a = a;
+  }
+
+  static make({
+    a,
+  }: {
+    a: number[];
+  }): result<Metalog, string> {
+    const validationResult = metalog.validate(a);
+    if(validationResult === metalog.MetalogValidationStatus.Success){
+      return Ok(new Metalog({a}))
+    }
+    else if(validationResult === metalog.MetalogValidationStatus.NotEnoughParamaters){
+      return Result.Error("Terms array must have more than one element");
+    }
+    else {
+      return Result.Error("Invalid combination of a (pdf is negative in parts)");
+    }
+  }
+  static fitFromCdf(
+    points: {x: number, q:number}[],
+    terms?: number
+  ): result<Metalog, string> {
+    const result = Metalog.fitFromCdfOLS(points, terms)
+    if(result.ok){
+      return result
+    }
+    else {
+      return Metalog.fitFromCdfLP(points, terms)
+    }
+  }
+
+  static fitFromCdfLP(
+    points: {x: number, q:number}[],
+    terms?: number
+  ): result<Metalog, string> {
+    let termCount = terms ?? points.length
+    if (termCount > 1) {
+      const a = metalog.fitMetalogLP(points.map(({x, q}) => ({x, y: q})), termCount);
+      if(a !== undefined){
+        return Ok(new Metalog({a }))
+      } else {
+        return Result.Error("Could not fit metalog from CDF points");
+      }
+    }
+    else {
+      return Result.Error("CDF must have more than 1 point");
+    }
+  }
+
+  static fitFromCdfOLS(
```

Comment:
This is almost identical to `fitFromCdfLP` above, would be cleaner to have a single function with the `exact` boolean parameter.